### PR TITLE
[DOCS] Create top-level "Search your data" page

### PR DESCRIPTION
**Pull request by jrodewig**
**Goal**

Create a top-level search section. This will let us clean up our search
API reference docs, particularly content from [`Request body search`][0].

**Changes**

* Creates a top-level `Search your data` page. This page is designed to
  house concept and tutorial docs related to search.

* Creates a `Run a search` page under `Search your data`. For now, This
  contains a basic search tutorial. The goal is to add content from
  [`Request body search`][0] to this in the future.

* Relocates `Long-running searches` and `Search across clusters` under
  `Search your data`. Increments several headings in that content.

* Reorders the top-level TOC to move `Search your data` higher. Also
  moves the `Query DSL`, `EQL`, and `SQL access` chapters immediately
  after.

Relates to #48194

As this effort will likely involve several redirects, I plan to merge this to `master` only.

[0]: https://www.elastic.co/guide/en/elasticsearch/reference/master/search-request-body.html

### CODE REVIEW

File: docs/reference/search/run-a-search.asciidoc [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+// TEST[continued]
+// TEST[s/^/PUT user_logs_000002\n/]
+====
+
+You can also search multiple indices using a glob or wildcard expression.
+
+.*Example*
+[%collapsible]
+====
+The following request uses the glob `user_logs*` in place of the index name.
```
Comment (debadair) [inline docs/reference/search/run-a-search.asciidoc:231]: This isn't specifically related to this reorg, but I've been wondering about the use of "glob" vs "index pattern". Given the prevalence of index pattern in other contexts, it feels like the glob terminology is potentially confusing. Glob isn't wrong in this context, but I'd be inclined to use index pattern, anyway.

**Review by debadair — APPROVED** [review]
LGTM.

File: docs/reference/search/run-a-search.asciidoc (hunk 1 above)
Comment (jrodewig) [reply docs/reference/search/run-a-search.asciidoc]: Solid point. Removed `glob` and used `index pattern` throughout. Thanks @debadair!

File: docs/reference/search/run-a-search.asciidoc [hunk 2]
```diff
@@ -0,0 +1,285 @@
...
+  simpler and best suited for testing.
+
+<<run-request-body-search,Request body searches>>::
+  Queries are provided through the JSON body of the API request. These queries
+  are written in <<query-dsl,Query DSL>>. We recommend using request body
+  searches in most production use cases.
+
+[WARNING]
+====
+If you specify both a query in both the URI and request body, the search API
```
Comment (lockewritesdocs) [inline docs/reference/search/run-a-search.asciidoc:21]: Suggest removing the first "both" to read:
> If you specify a query in both the URI and request body...

File: docs/reference/search/run-a-search.asciidoc [hunk 3]
```diff
@@ -0,0 +1,285 @@
...
+
+GET /*/_search
+{
+    "query" : {
+        "match" : { "message" : "login" }
+    }
+}
+----
+// TEST[continued]
+====
```
Comment (lockewritesdocs) [inline docs/reference/search/run-a-search.asciidoc:285]: The two previous examples have a lead-in sentence that describes the collapsed example. Just noting that this example does not follow that convention. 

image

**Review by lockewritesdocs — APPROVED** [review]
Two minor suggestions, but approved otherwise.

File: docs/reference/search/run-a-search.asciidoc (hunk 2 above)
Comment (jrodewig) [reply docs/reference/search/run-a-search.asciidoc]: Good catch. Fixed with d05ea62df90.

File: docs/reference/search/run-a-search.asciidoc (hunk 3 above)
Comment (jrodewig) [reply docs/reference/search/run-a-search.asciidoc:285]: I think I'm okay with that in this case. Thanks for raising though!

**Comment by jrodewig** [comment]
### Backport commits
master a73fef3
7.x 50a8779
7.8 b25146a 
7.7 37633a4

**Pull request merged**